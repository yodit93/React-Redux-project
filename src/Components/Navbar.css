/* eslint-disable */
/* stylelint-disable */
* {
  padding: 0;
  margin: 0;
}

.logo {
  display: flex;
  align-items: center;
  margin-left: 2rem;
}

.logo h1 {
  font-size: 2rem;
}

.logo-image {
  height: 5rem;
  padding-right: 3rem;
}

