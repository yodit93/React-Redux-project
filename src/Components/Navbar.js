import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import logo from '../logo/planet.png';
import './Navbar.css';

function Navbar() {
  return (
    <nav className="navbar">
      <div className="logo">
        <img className="logo-image" src={logo} alt="Logo" />
        <h1>Space Travelers` Hub</h1>
      </div>

      <div className="nav-links">
        <ul className="menu-list">
          <li className="list-item">
            <Link to="/">Rockets</Link>
          </li>
        </ul>
      </div>
      <NavLink />
    </nav>
  );
}

export default Navbar;
