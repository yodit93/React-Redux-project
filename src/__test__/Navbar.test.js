import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Navbar from '../components/Navbar';

const MockNavbar = () => (
  <BrowserRouter>
    <Navbar />
  </BrowserRouter>
);

test('should render the Navbar component', () => {
  const tree = renderer.create(<MockNavbar />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should include nav element', () => {
  render(<MockNavbar />);
  const navElement = screen.getByRole('navigation');
  expect(navElement).toBeInTheDocument();
});
