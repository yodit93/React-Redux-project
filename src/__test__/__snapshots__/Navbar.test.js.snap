// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Navbar component 1`] = `
<nav
  className="navbar"
>
  <div
    className="logo"
  >
    <img
      alt="Logo"
      className="logo-image"
      src="planet.png"
    />
    <h1>
      Space Travelers\` Hub
    </h1>
  </div>
  <div
    className="nav-links"
  >
    <ul
      className="menu-list"
    >
      <li
        className="list-item"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="list-item"
      >
        <a
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="list-item"
      >
        <a
          href="/profile"
          onClick={[Function]}
        >
          Profile
        </a>
      </li>
    </ul>
  </div>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  />
</nav>
`;
